src/vs/code/node/cli.ts

```diff
@@ -319,23 +324,61 @@ export async function main(argv: string[]): Promise<any> {
 			options['stdio'] = 'ignore';
 		}
 
-		const child = spawn(process.execPath, argv.slice(2), options);
+		let child: ChildProcess;
+		if (platform() !== 'darwin') {
+			child = spawn(process.execPath, argv.slice(2), options);
 
-		if (args.wait && waitMarkerFilePath) {
-			return new Promise<void>(resolve => {
+			if (args.wait && waitMarkerFilePath) {
+				return new Promise<void>(resolve => {
 
-				// Complete when process exits
-				child.once('exit', () => resolve(undefined));
+					// Complete when process exits
+					child.once('exit', () => resolve(undefined));
 
-				// Complete when wait marker file is deleted
-				whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
-			}).then(() => {
+					// Complete when wait marker file is deleted
+					whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
+				}).then(() => {
 
-				// Make sure to delete the tmp stdin file if we have any
-				if (stdinFilePath) {
-					unlinkSync(stdinFilePath);
-				}
-			});
+					// Make sure to delete the tmp stdin file if we have any
+					if (stdinFilePath) {
+						unlinkSync(stdinFilePath);
+					}
+				});
+			}
+		} else {
+			const requiresWait = verbose || hasReadStdinArg;
+			const openArgs: string[] = ['-n'];
+
+			let tmpStdoutLogger: CliVerboseLogger | undefined;
+			let tmpStderrLogger: CliVerboseLogger | undefined;
+			if (requiresWait) {
+				const tmpStdoutName = createFileName(tmpdir(), 'code-stdout');
+				const tmpStderrName = createFileName(tmpdir(), 'code-stderr');
+				tmpStdoutLogger = new CliVerboseLogger(tmpStdoutName);
+				tmpStderrLogger = new CliVerboseLogger(tmpStderrName);
+				openArgs.push('-W');
+				openArgs.push('--stdout', tmpStdoutName);
+				openArgs.push('--stderr', tmpStderrName);
+			}
+			const argsArr: string[] = [];
+			const isDev = env['NODE_ENV'] === 'development';
+			// When we're in development mode, call the OSS app rather than the OSS app's Electron
+			const execPathToUse = isDev ? resolve(join(process.execPath, '../../..')) : process.execPath;
```

Comment:
What is the difference between

`open -a <path-to>/Code\ -\ OSS.app` and `open -a <path-to>/Code\ -\ OSS.app/Contents/MacOS/Electron`

I would expect both to be the same, the main difference for OSS is what follows `--args` which should be path to vscode root folder.

Reply:
That is not related to the executable path but rather the path pointed to electron application source. Can you provide example of failing call.

Reply:
You're right. It was pointing to the app file instead of the vscode directory because I tried using the longer `process.execPath`, but didn't change line 373.